src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -422,6 +422,7 @@ public interface IStartConfiguration
             IAuthProvider AuthProvider { get; set; }
             IEnumerable<IExtension> Extensions { get; set; }
             TaskProcessMode ProcessMode { get; set; }
+            bool HeadlessMode { get; set; }
```

Comment:
@ikeough @gregmarr , this is a change to a public interface, any change to this interface will break clients building their own Start Configuration derived configurations... like Revit etc, and should impact Dynamo version number to Dynamo 2.0 afaik.

I think an alternative is to implement an additional interface containing the headless mode and make `DefaultStartConfig` inherit from that as well.

We should be consistent on how we handle these interface additions throughout the code I think. tagging @Benglin because we were discussing this.

for example, adding `ITransformable` instead of modifying `IRenderPackage` or making `IRenderPackage 1.1`.

I guess we could also just implement the change in the type and not touch the interface.